legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/relationalTypeTranslations.pure

```diff
@@ -0,0 +1,27 @@
+import meta::relational::runtime::*;
+import meta::pure::metamodel::type::*;
+import meta::relational::metamodel::DataType::*;
+import meta::relational::translation::*; 
+
+Class meta::relational::translation::TranslationContext
+{     
+   dbType: DatabaseType[0..1];
+}
+
+function meta::relational::translation::translateCoreTypeToDbSpecificType(type: meta::relational::metamodel::datatype::DataType[0..1], translationContext : TranslationContext[1]): meta::relational::metamodel::datatype::DataType[0..1]
+{
+   if($type->isEmpty() || $translationContext.dbType->isEmpty(), |$type,
+      |let translatedType = translateCoreTypeToDbSpecificTypeMap($type->toOne())->get($translationContext.dbType->toOne());
+      if($translatedType->isEmpty(), |$type, |$translatedType););
+}
+
+function <<access.private>> meta::relational::translation::translateCoreTypeToDbSpecificTypeMap(type: meta::relational::metamodel::datatype::DataType[1]):Map<DatabaseType,meta::relational::metamodel::datatype::DataType>[1]
+{   
```

Comment:
This function has db specific logic. Shouldn't db specific logic be in db specific abstractions. 

See my comment in #465

Reply:
yes, this is being discussed with pierre/david, on how should we refactor this, as this stuff is spread across platform and core_relational